controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -393,23 +382,26 @@ public OperationContext createContext(String scope, String name) {
         return withCompletion(stream.scaleTryDeleteEpoch(epoch), executor)
                 .thenCompose(deleted -> {
                     if (deleted) {
-                        return stream.latestScaleData()
-                                .thenCompose(pair -> {
-                                    List<Integer> segmentsSealed = pair.getLeft();
-                                    return FutureHelpers.allOfWithResults(pair.getRight().stream().map(stream::getSegment)
-                                            .collect(Collectors.toList()))
-                                            .thenApply(segmentsCreated ->
-                                                    new DeleteEpochResponse(true, segmentsSealed, segmentsCreated));
-                                });
+                        return stream.latestScaleData().thenCompose(pair -> {
+                            List<Integer> segmentsSealed = pair.getLeft();
+                            CompletableFuture<List<Segment>> segmentsCreated = FutureHelpers.allOfWithResults(
+                                    pair.getRight().stream().map(stream::getSegment).collect(Collectors.toList()));
+                            CompletableFuture<Long> scaleTimestamp = pair.getRight().isEmpty() ?
+                                    stream.getSegment(segmentsSealed.stream().max(Comparator.naturalOrder()).get())
+                                            .thenApply(Segment::getStart) :
+                                    segmentsCreated
+                                            .thenApply(x -> x.get(0).getStart());
+                            return scaleTimestamp.thenApply(ts -> new DeleteEpochResponse(true, segmentsSealed, ts, segmentsCreated.join()));
+                        });
                     } else {
                         return CompletableFuture.completedFuture(
-                                new DeleteEpochResponse(false, null, null));
+                                new DeleteEpochResponse(false, null, 0, null));
                     }
                 });
     }
 
     @Override
-    public CompletableFuture<VersionedTransactionData> createTransaction(final String scopeName,
+    public CompletableFuture<Pair<VersionedTransactionData, List<Segment>>> createTransaction(final String scopeName,
```

Comment:
List of segments is later used to create txn segments on relevant segment stores. Sending the list here avoids additional store call to fetch list of segments active in the epoch.